src/overlays/actors/ovl_Bg_Lotus/z_bg_lotus.c

```diff
@@ -20,22 +23,141 @@ const ActorInit Bg_Lotus_InitVars = {
+}
+
+void BgLotus_Destroy(Actor* thisx, PlayState* play) {
+    BgLotus* this = THIS;
+
+    DynaPoly_DeleteBgActor(play, &play->colCtx.dyna, this->dyna.bgId);
+}
+
+void func_80AD6830(BgLotus* this) {
+    f32 temp_fv1;
+
+    if (this->dyna.actor.params == 0) {
+        temp_fv1 = Math_SinF(this->unk166 * (M_PI / 4)) * ((0.014f * (this->unk166 / (80.0f * 0.1f))) + 0.01f);
+        this->dyna.actor.scale.x = ((1.0f + temp_fv1) * 0.1f);
+        this->dyna.actor.scale.z = ((1.0f - temp_fv1) * 0.1f);
+    }
+}
+
+void func_80AD68DC(BgLotus* this, PlayState* play) {
+    f32 sp34;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Lotus/BgLotus_Init.s")
+    this->unk168--;
+    sp34 = Math_SinF(this->unk168 * 0.06544985f) * 6.0f;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Lotus/BgLotus_Destroy.s")
+    if (this->dyna.actor.params == 0) {
+        this->dyna.actor.world.pos.x = (Math_SinS(this->dyna.actor.world.rot.y) * sp34) + this->dyna.actor.home.pos.x;
+        this->dyna.actor.world.pos.z = (Math_CosS(this->dyna.actor.world.rot.y) * sp34) + this->dyna.actor.home.pos.z;
+        if (this->unk168 == 0) {
+            this->unk168 = 0x60;
+            this->dyna.actor.world.rot.y += (s16)((s32)Rand_Next() >> 0x12);
+        }
+    }
+    if (this->unk160 < this->dyna.actor.floorHeight) {
+        this->dyna.actor.world.pos.y = this->dyna.actor.floorHeight;
+    } else {
+        this->dyna.actor.world.pos.y = this->unk160;
+        if (DynaPolyActor_IsPlayerOnTop(&this->dyna)) {
+            if (!this->unk164) {
+                EffectSsGRipple_Spawn(play, &this->dyna.actor.world.pos, 0x3E8, 0x578, 0);
+                EffectSsGRipple_Spawn(play, &this->dyna.actor.world.pos, 0x3E8, 0x578, 8);
+                this->unk166 = 40;
+            }
+            if (gSaveContext.save.playerForm != PLAYER_FORM_DEKU) {
+                this->unk166 = 40;
+                this->dyna.actor.flags |= ACTOR_FLAG_10;
+                this->actionFunc = func_80AD6A88;
+                return;
+            }
+            this->unk164 = true;
+        } else {
+            this->unk164 = false;
+        }
+    }
+    if (this->unk166 > 0) {
+        this->unk166--;
+    }
+    func_80AD6830(this);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Lotus/func_80AD6830.s")
+void func_80AD6A88(BgLotus* this, PlayState* play) {
+    if (this->unk160 < this->dyna.actor.world.pos.y) {
+        this->dyna.actor.world.pos.y = this->unk160;
+    }
+    this->dyna.actor.world.pos.y -= 1.0f;
+    if (this->dyna.actor.world.pos.y <= this->dyna.actor.floorHeight) {
+        this->dyna.actor.world.pos.y = this->dyna.actor.floorHeight;
+        this->unk166 = 0;
+    }
+    if (this->unk166 > 0) {
+        this->unk166--;
+        func_80AD6830(this);
+        return;
+    }
+    if (Math_StepToF(&this->dyna.actor.scale.x, 0.0f, 5.0f * 0.001f)) {
+        this->dyna.actor.draw = NULL;
+        this->unk166 = 0x64;
```

Comment:
```suggestion
        this->unk166 = 100;
```
`unk166` seems to be a timer, so this should be in decimal